### PR TITLE
Cult Improvement Drive

### PR DESCRIPTION
**Pull request by ghost**
This PR aims to make cult more engaging by changing rune functionalities, introducing new features, and changing objectives.

Changes:
- Cult structures now have a set amount of health and will break when destroyed
- Cult floors will now heal any cultists standing on them for 1 brute and 1 burn per tick.
- Tomes can now only be used when you are conscious, standing, etc.
- Cultists can no longer use null rods to erase runes
- Blind and Deafen talismans have been re-added
- Hide and Reveal Runes talismans now have 3 uses
- EMP runes are a new color (bluish-violet)
- The Corrupt rune has been added; upon use, it will cultify nearby turfs and transform any metal sheets into runed metal sheets

### CODE REVIEW

**Comment by Bawhoppen** [comment]
#ooh #ah

File: code/game/gamemodes/cult/cult_structures.dm [hunk 1]
```diff
@@ -1,36 +1,139 @@
...
 	icon_state = "hole"
 	density = 1
 	unacidable = 1
 	anchored = 1
+
+////////////////
+// CULT TURFS //
+////////////////
+
+/turf/simulated/floor/plasteel/cult
```
Comment (Anonus) [inline code/game/gamemodes/cult/cult_structures.dm]: You need to do this for `/turf/simulated/floor/engine/cult` as well, or preferably remove one of those paths.

**Comment by KorPhaeron** [comment]
>Blind and Deafen talismans have been re-added

The reason this was removed is because a cultist being able to have a bag of 50 aoe "fuck you, you lose" papers was not fun. It won't be fun this time either.

**Comment by Xhuis** [comment]
@KorPhaeron The only one that cripples nearby people is the Blind talisman, which only blinds for ten seconds.

**Comment by KorPhaeron** [comment]
10 seconds of being unable to fight is gigantic. Thats twice the lenght of most stuns. It's aoe and entirely unblockable outside of a single item

And you can stack a ton of them in your bag so really its 100 seconds.

Trust me, we tried this, it was abused to hell, we removed it.

**Comment by Metacide** [comment]
The rest sounds really neat though. Would be nice to see a dramatic fight over a properly-styled cult base.

**Comment by Xhuis** [comment]
A lot of different runes are going to require certain structures. For instance, the Sacrifice rune now requires two pylons and an altar nearby.

**Comment by Zilenan91** [comment]
Xhuis, please, close this.

I'd offer criticism in any way I possibly could, but there isn't any. It's not that your changes are bad, but that cultists are such a mess of design, that by adding more and more nebulous "stuff" for them to do will just make the mode more unbearable than it already is. 

You put in base-building mechanics but don't remove their escape or mobility abilities. You add in blind and deafen talismans which were absolute cancer before and will be cancer now. You add in "cult weeds" to make cult autism forts even stronger and yet more unassailable due to the spamming of stun paper. But above all you do absolutely nothing about said stun paper, something that I'm honestly going to say doesn't just kill my enjoyment of cult mode, but kills my enjoyment of the entire game as a whole because of its sheer influence making me not want to play a majority of the jobs I'd enjoy.

**Comment by Xhuis** [comment]
@Zilenan91 Due to your complete failure to read the fact that this is _work in progress_ and that _everything is subject to change_, I'm going to completely fail to read your post.

**Comment by Zilenan91** [comment]
@Xhuis If it's WiP, why isn't there a Work in Progress tag

**Comment by KorPhaeron** [comment]
Xhuis can't add tags, only maintainers can

**Comment by Zilenan91** [comment]
@KorPhaeron Huh. I thought that creators of pull requests would be able to add their own tags for organization purposes. That's really bad design on githubs side.

**Comment by duncathan** [comment]
@Zilenan91 that's what the big (WIP) in the PR title is for

**Comment by Xhuis** [comment]
New in this commit:
* By grabbing someone and clicking on a forge, you can DUNK THEIR FUCKING FACE INTO LAVA.
* Smithing! You can add an object to a forge and alt-click to attempt to modify it. For instance, turn a bedsheet into a cult version.
* cult_items.dm and cult_structures.dm have been moved into a new file under the cult folder, objects.dm.
* Cult walls can no longer be deconstructed with a welding tool
* Cult floors and walls can now be purified with a null rod
* Runed metal is now obtainable from the supply talisman in stacks of 20
* Many runes now require certain structures to work
* Boil Blood now deals 75 fire damage to non-cultists in addition to setting them on fire with 5 fire stacks
* Admin-only imbued tomes now exist for instantly drawing runes
* You can smith a bible to create an accursed tome, which draws runes one second faster
* Null rod wearers can no longer be harmed by tome-smacking

**Comment by Anonus** [comment]
Cult walls should probably have some way for the cult to remove or move stuff through them.

**Comment by Xhuis** [comment]
@Anonus That's a good point. I might make them removable via tomes.

File: code/game/gamemodes/cult/runes.dm [hunk 2]
```diff
@@ -92,8 +94,10 @@ structure_check() searches for nearby cultist structures required for the invoca
 /obj/effect/rune/proc/can_invoke(var/mob/living/user)
 	//This proc determines if the rune can be invoked at the time. If there are multiple required cultists, it will find all nearby cultists.
 	if(!structure_check(req_pylons, req_forges, req_archives, req_altars))
+		user << "<span class='warning'>[src] lacks the nearby structures it requires!</span>"
 		return 0
 	if(!keyword && req_keyword)
+		user << "<span class='warning'>[src] requires no keyword!</span>"
```
Comment (Anonus) [inline code/game/gamemodes/cult/runes.dm]: requires a keyword, not no keyword.
though given that this won't normally appear, eh.

File: code/game/turfs/simulated/floor/plasteel_floor.dm [hunk 3]
```diff
@@ -370,10 +370,6 @@
 /turf/simulated/floor/plasteel/vault
 	icon_state = "vault"
 
-/turf/simulated/floor/plasteel/cult
```
Comment (Anonus) [inline code/game/turfs/simulated/floor/plasteel_floor.dm]: You need to remove lines 382 to 388, at least if that won't cause map errors.

Comment (Xhuis) [inline code/game/turfs/simulated/floor/plasteel_floor.dm]: It will cause map errors, hence why I kept it in for legacy purposes.

**Comment by Xhuis** [comment]
Cuboos has made a potential new Nar-Sie sound. Thoughts?
https://dl.dropboxusercontent.com/u/9654207/NarsieRisen.wav

**Comment by Anonus** [comment]
Gotta convert nar-sie's tile replacement and the artificer spell to use `/turf/simulated/floor/engine/cult` but you probably know that.

**Comment by Bawhoppen** [comment]
@Xhuis people are going to hate that sound, people will want to keep the old one
we saw how well the other sound went

**Comment by Nitrousoxide** [comment]
>Cuboos has made a potential new Nar-Sie sound. Thoughts?
>https://dl.dropboxusercontent.com/u/9654207/NarsieRisen.wav

That sound is about 7 seconds too long.

**Comment by Xhuis** [comment]
Nar-Sie spawns about five seconds into the sound.

**Comment by Xhuis** [comment]
New in this commit:
* The spell Lesser Construction now creates engine cult turf, as does Nar-Sie
* The tome menu now has a Cancel option
* Nar-Sie has a new sound
* The rune for summoning Nar-Sie now becomes distorted and shifting when the summoning begins (see here: http://i.imgur.com/Z6fA6Cm.gifv)

**Comment by Robustin** [comment]
I dont care about blind/deafen, I've literally never seen them used and I'm one of the most experienced cultists OF ALL TIME, OF ALL TIME!

So I don't know why anyone's having a hissy fit over it when cult still has/had an AOE deathbomb and single target i-win talismans. Similarly, dont know why it should be re-added either, would just clutter up rune menu, maybe merge them into a blind+deaf rune (Call it Overawe or something pimp).

**Comment by KorPhaeron** [comment]
>I've literally never seen them used

Take it from someone who has then. "You enter the same screen as a cultist and you lose" is not fun gameplay. There is literally no way to stop the blind outside of a single item. You can't fight at all. One cultist can take out the entire escape shuttle or all of security on their own because they will be blind for the whole fight. Blood boil requires a stationary rune and three people. Stun talisman requires you to melee them. Blind paper requires nothing.

We just nerfed AoE flashes for being the same nonsense and those actually had counters. 

I don't object to this update in general but it's frustrating to see you pushing ahead with the same mistakes (blind talisman, instant walls that are hard to get rid of) we already made, tested, and corrected after months/years of feedback on the game. Why do we have to go through the same trial and error again?

**Comment by Xhuis** [comment]
By "instant walls", are you referring to wall runes? I can easily nerf those - perhaps make them return to normal after a minute or two.

Blind and Deafen talismans are staying, but may be modified drastically. I'm leaning towards making them melee and single-target with more powerful effects, similar to the Stun talisman.

**Comment by KorPhaeron** [comment]
Single target would be fine though redundant, or making flash protection stop them (or do eyeblur only?) so they can be used on crowds of validhunters but not instantly cripple security

**Comment by Zilenan91** [comment]
@xhuis How about you make it so the unweldable cult walls require a corruption rune next to them to maintain their cult status? 
Activate the corruption rune to make them cult, slap another one on the floor to maintain their power.

**Comment by Xhuis** [comment]
New in this commit:
* New pylon breaking sound by Cuboos
* Fixed an unreported issue where the one-use talisman to summon tomes said it had two uses

**Comment by Xhuis** [comment]
New in this commit:
* Cult floors and walls can now be deconstructed, yielding runed metal when destroyed.
* Hitting a cult floor or wall with a bible will weaken it and make it susceptible to deconstruction.
* Cultists cannot perform this interaction for obvious reasons as well as the ability to duplicate runed metal.

**Comment by Xhuis** [comment]
This can be considered merge-ready once I push the changelog, but I want a few test rounds ran before this gets merged.

**Comment by Aranclanos** [comment]
>This is DNM for now.

**Comment by Xhuis** [comment]
@Aranclanos I forgot to update the OP. This isn't WIP anymore, nor is it DNM. I still want to run some test rounds before it gets in, though.

**Comment by Xhuis** [comment]
New objective added: Bring the Slaughter. After invoking the new Call Forth The Slaughter rune, three cult-aligned, empowered slaughter demons will be summoned to kill the heretics still alive aboard the station. They also have the ability to locate any living beings that aren't cultists.

File: code/game/gamemodes/cult/cult.dm [hunk 4]
```diff
@@ -54,15 +54,16 @@
 	antag_flag = ROLE_CULTIST
 	restricted_jobs = list("Chaplain","AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel")
 	protected_jobs = list()
-	required_players = 20
-	required_enemies = 4
-	recommended_enemies = 4
+	required_players = 1 //20
+	required_enemies = 1 //4
+	recommended_enemies = 1 //4
```
Comment (Xhuis) [inline code/game/gamemodes/cult/cult.dm]: oops debug code, one sec

**Comment by Zilenan91** [comment]
Can you make it so the demons have to kill the cultists too? Flavorful since nar-sie prioritizes cultists over regular crew.

**Comment by Xhuis** [comment]
I removed the debug code. I also fixed an entirely unrelated issue where headslugs didn't have the right death icon.

@Zilenan91 I debated this but decided against it, as it might discourage the cultists from actually doing the objective until the last second, and it doesn't seem like it'd be very fun for the cultists who toiled to actually bring it about.

**Comment by Zilenan91** [comment]
@Xhuis but Nar-sie already kills everyone, I don't think the cultists are gonna give two shits if they die.

**Comment by Xhuis** [comment]
Nar-Sie returns the cultists and everyone else as harvesters.

**Comment by Zilenan91** [comment]
In my experience being a harvester, while fun, is usually a complete waste of time. You just kinda run at dudes and GGNORE them with insta inject gas clouds of permastun.

**Comment by Bawhoppen** [comment]
its end of round intentionally OP

**Comment by Xhuis** [comment]
New in this commit:
* Stun talismans now operate off of a subtype of /talisman, /melee
* Blind and Deafen talismans are now melee talismans and have enhanced effects against single targets

File: code/game/gamemodes/cult/cult.dm [hunk 5]
```diff
@@ -76,9 +77,12 @@
 
 
 /datum/game_mode/cult/pre_setup()
-	if(prob(50))
+	if(prob(33))
 		cult_objectives += "survive"
 		cult_objectives += "sacrifice"
+	if(prob(50))
```
Comment (Anonus) [inline code/game/gamemodes/cult/cult.dm]: Wouldn't this have a chance of giving a cult four objectives?

Comment (Xhuis) [inline code/game/gamemodes/cult/cult.dm]: Good point. I'll hotfix that.

**Comment by Xhuis** [comment]
I just decided to straight-up rewrite how cult objectives are selected. It now chooses from a weighted list, with Escape having a probability of 20% - Nar-Sie and Slaughter both have a 40% chance.

**Comment by Xhuis** [comment]
I'll squash commits in just a sec, but the minimum is 3 because of the merge conflict resolution.

**Comment by Xhuis** [comment]
Commits squashed to as few as they'll go. If it's still a problem I can manually copy over everything to a new branch and have only one commit that way.

**Comment by Xhuis** [comment]
This still needs earlymerge testing before it gets fully merged.

**Comment by Xhuis** [comment]
New in these commits:
* Harbringers of the Slaughter now have a large name in cult communications
* Captains can now be converted to cult (but still must have their loyalty implant removed first)
* The convert rune now knocks out the target, instead of the conversion itself

**Comment by KorPhaeron** [comment]
Wait what why are slaughter demons in cult

**Comment by Xhuis** [comment]
People thought they were thematically fitting, and it seemed true enough to me. They're only summoned as a round-ending objective.

**Comment by KorPhaeron** [comment]
Well I guess its not a balance issue if they're round end but I always imagined them unaffiliated with nar sie.

I'll put this in rotation/test merge it in a little while.

**Comment by Dorsisdwarf** [comment]
I don't like that captains can be converted now.
I don't like converting the captain in any mode frankly

**Comment by GunHog** [comment]
Why? No one should be *immune* to conversion except Chaplain, as he is protected by...god magic? If you can capture the Captain alive, you deserve to have that prize on your side, on par with managing to snag the HoS.

**Comment by Dorsisdwarf** [comment]
Because the captain was always the one inconvertible guy, the one who always leads the defense, and if he was any good he was the last man standing. Now he'll only immune in 1/4 conversion modes, which is ew.

Hell, a few months ago the HOP was being considered for hardcoded cult immunity (His blandness of soul is too great even for the Geometer to find palatable)

**Comment by GunHog** [comment]
HoP was removed from *starting* as cult. The reason you do not let HoP start as cult is because of all-access. Killing the HoP and taking his ID to spread access is not that much different than converting him for that same purpose.

**Comment by Xhuis** [comment]
Melee talismans, such as stun, are now creatable again.

**Comment by adrix89** [comment]
Captain has always had brainwashing protection training :-1:
Do you think Nanotrasen will have the captain be a liability?

**Comment by Xhuis** [comment]
Yes.

**Comment by Anonus** [comment]
Artificers can't use the floor or wall spells on centcom, which used to be needed due to them being able to crash the server but should no longer be needed now that the transit areas work differently.

**Comment by Zilenan91** [comment]
So it's all ogre now?

**Comment by Anonus** [comment]
Well, if this is closed I guess I'll do my thing and give constructs cool action buttons and add a few cool effects.

**Comment by Zilenan91** [comment]
@Anonus It'd be best if you didn't, admins have been removing cult from rotation lately because it's just not good. Adding more things to it and buffing it won't solve the core problem of it being quite literally an abortion of design due to its creator leaving, and everything after that being tacked on.

**Comment by Anonus** [comment]
I won't be buffing it, just slapping cool effects on it because effects are cool, and leaving constructs without action buttons is a tragedy.

**Comment by duncathan** [comment]
@Zilenan91 did cult mode kill your parents or something

**Comment by Zilenan91** [comment]
@duncathan 
Yes

But for real, Cult is awful because at the core of it, it's a bunch of dudes who spawn with and can make a functionally infinite amount of parapens, gib people at will, summon unstunnable death bots, can 1-shot crit entire rooms at no effort, have a dedicated AI killer... I could go on. It has so many things about it that it just does not need and it feels very cheap.

**Comment by duncathan** [comment]
@Zilenan91 personally I think that the core concept isn't flawed; it's the execution. It would take a lot of effort to truly do it well, though.

**Comment by Zilenan91** [comment]
Yeah, a total rework. That's why cult isn't able to be feasibly saved, too much work for anyone to ever really want to do. 

But yeah I meant core gameplay, not core concept, my bad.

**Comment by Anonus** [comment]
I'm commenting here so I can link a screenshot to IRC because my internet is being really finicky.
screenshot_18

**Comment by KorPhaeron** [comment]
Good stuff. What happened to this PR though? It's on server right now

**Comment by Zilenan91** [comment]
It's still under testing I think, never got merged.

**Comment by Xhuis** [comment]
I'm no longer ridiculously frustrated.

**Comment by Zilenan91** [comment]
What in the fuck

**Comment by Iamgoofball** [comment]
why did you quit xhuis i was looking forward to this

**Pull request closed without merging**